src/test/java/com/malinatran/request/RequestListenerTest.java

```diff
@@ -0,0 +1,21 @@
+package com.malinatran.request;
+
+import com.malinatran.reader.Reader;
+import com.malinatran.mocks.MockRequestReader;
+
+import java.io.IOException;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class RequestListenerTest {
+
+    @Test
+    public void testGetNextRequest() throws IOException {
+        Reader rr = new MockRequestReader(new String[]
```

Comment:
For your tests, could you put newlines between your test setup, invocation, and assertions? Makes it easier to read